spec/requests/api/blueprints_spec.rb

```diff
@@ -0,0 +1,111 @@
+RSpec.describe "Blueprints API" do
+  class Blueprint
+    def self.find(*)
+      all.first
+    end
+
+    def self.create(attributes = {})
+      new.tap do |bp|
+        attributes.each { |attribute, value| bp.public_send("#{attribute}=", value) }
+        all << bp
+      end
+    end
+
+    def self.apply_legacy_finder_options(*)
+      all
+    end
+
+    def self.base_class
+      self
+    end
+
+    def self.all
+      @all ||= []
+    end
+
+    def self.count
+      all.count
+    end
+
+    attr_accessor :name, :description, :ui_properties
+
+    def id
+      1
+    end
+
+    def save; end
+
+    def [](attribute)
+      public_send(attribute) rescue nil
+    end
+
+    def attributes
+      {:name => name, :description => description, :ui_properties => ui_properties}
+    end
+
+    def create_bundle(args)
+      # NOP
+    end
+  end
+
+  after { Blueprint.instance_variable_set(:@all, nil) }
+
+  it "lists all the blueprints" do
+    Blueprint.create
+    api_basic_authorize
+
+    run_get(blueprints_url)
+
+    expected = {
+      "count" => 1,
+      "subcount" => 1,
+      "name" => "blueprints",
+      "resources" => Blueprint.all.map { |bp| hash_including("href" => a_string_matching(blueprints_url(bp.id))) }
+    }
+    expect(response_hash).to include(expected)
+    expect(response).to have_http_status(:ok)
+  end
+
+  it "can create a blueprint" do
+    api_basic_authorize
+
+    run_post(blueprints_url, {:name => "foo", :description => "bar", :ui_properties => {:some => {:json => "baz"}}})
+
+    expected = {
+      "results" => [
+        {
+          "name" => "foo",
+          "description" => "bar",
+          "ui_properties" => {"some" => {"json" => "baz"}}
+        }
+      ]
+    }
+    expect(response_hash).to include(expected)
+    expect(response).to have_http_status(:ok)
+  end
+
+  it "can create a bundle on a blueprint" do
+    blueprint = Blueprint.create
+    service_dialog = FactoryGirl.create(:dialog)
+    service_template_1, service_template_2 = FactoryGirl.create_list(:service_template, 2)
+    service_catalog = FactoryGirl.create(:service_template_catalog)
+    api_basic_authorize
+
+    run_post(
+      blueprints_url(blueprint.id),
+      {
```

Comment:
@abellotti I tried to find a way to form this JSON in a way that seems consistent with other endpoints....is this the best way to express these requirements? I'm very open to suggestions here! :musical_score: